### PR TITLE
Adapt to rmm logger changes

### PR DESCRIPTION
**Pull request by vyasr**
This PR adapts to breaking changes in rmm in https://github.com/rapidsai/rmm/pull/1722.

### CODE REVIEW

File: cpp/CMakeLists.txt
Comment (KyleFromNVIDIA) [inline cpp/CMakeLists.txt:191]:
```suggestion
    target_link_libraries(ucxx
      PUBLIC rmm::rmm rmm::rmm_logger
      PRIVATE $<TARGET_NAME_IF_EXISTS:rmm::rmm_logger_impl>
    )
```

File: ci/build_cpp.sh
Comment (pentschev) [inline ci/build_cpp.sh:23]: Do we want to merge with those here or wait for that PR to be merged before merging here?

Comment (vyasr) [reply ci/build_cpp.sh]: Definitely wait. The upstream PR is merged now.

**Review by KyleFromNVIDIA — APPROVED** [review]

**Pull request merged**
